packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -216,6 +216,16 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
     );
   }
 
+  /// Retrieves the challenge parameters that are not part of standard Cognito InitiateAuth flow.
+  Map<String, String>? _getAdditionalInfo(
+    Map<String, String> challengeParameters,
+  ) {
+    final additionalInfo = {...challengeParameters}..removeWhere(
```

Comment:
Not following why this is needed over just returning the challenge params

Reply:
I believe that the data in `state.challengeParameters` (and ultimately from the sdk response) include parameters for a variety of auth challenges that Cognito may issue. In addition, it can include the `publicChallengeParameters` returned by customer-written lambda code. My expectation is that this member wouldn't include things like `'SALT` and other values used internally by our library when calling Cognito, but other information which may be present in the response (which basically boils down to just the lambda response, but I'm trying to confirm this).

Reply:
In general, yes, but the value of `challengeParameters` for the CUSTOM_CHALLENGE state only has the username plus any custom values sent from the backend, e.g.

```json
{
  "USERNAME": "test4",
  "trigger": "true"
}
```

Was there a case where you noticed more than this being returned?

Reply:
Oh I see. I was just testing the size of the list for anything beyond the publicChallengeParameters configured by the customer in their lambda. If these are the only values being appended, we can probably just leave it for now.